org.eclipse.xpect.ui/src/org/eclipse/xpect/ui/preferences/XpectRootPreferencePage.java

```diff
@@ -12,22 +12,51 @@
 
 package org.eclipse.xpect.ui.preferences;
 
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
 import org.eclipse.core.resources.IProject;
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.layout.GridLayoutFactory;
 import org.eclipse.jface.preference.BooleanFieldEditor;
 import org.eclipse.swt.SWT;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
 import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Group;
 import org.eclipse.xtext.ui.editor.preferences.IPreferenceStoreAccess;
 import org.eclipse.xtext.ui.editor.preferences.LanguageRootPreferencePage;
 import org.eclipse.xpect.ui.internal.XpectActivator;
 
 public class XpectRootPreferencePage extends LanguageRootPreferencePage {
 
 	public static final String LIVE_TEST_EXECUTION_PREFERENCE_NAME = "org.eclipse.xpect.ui.live_test_execution";
+	
+	public static final String SKIP_CONTENT_CHECK_PREFERENCE_NAME = "org.eclipse.xpect.ui.skip_content_check";
 
+	public static final String DISABLE_EDITOR_OVERRIDE_PREFERENCE_NAME = "org.eclipse.xpect.ui.editor_override";
+	
 	@Override
 	protected void createFieldEditors() {
 		super.createFieldEditors();
 		Composite parent = getFieldEditorParent();
+		if (!isPropertyPage()) {
+			Group group = new Group(parent, SWT.NONE);
+			group.setText("Xpect Editor");
+			GridLayoutFactory.fillDefaults().margins(5, 5).applyTo(group);
+			GridDataFactory.fillDefaults().span(2, 1).grab(true, false).applyTo(group);
+			
+			Composite skipFileExtListComposite = new Composite(group, SWT.NONE);
+			GridLayoutFactory.fillDefaults().margins(5, 5).applyTo(skipFileExtListComposite);
+			GridDataFactory.fillDefaults().grab(true, false).applyTo(skipFileExtListComposite);
+			addField(new SkipFileExtensionList(LIVE_TEST_EXECUTION_PREFERENCE_NAME, "Skip content check for XPECT for extensions (will not be used if override is disabled):", skipFileExtListComposite));
```

Comment:
I think we can skip the "will not be used if override is disabled", there is an overlap of the 2 preferences but I doubt the user cares.

Maybe you can disable the entire group with the settings per preference, if the checkbox for the global disable is ticket. A nice to have though, I don't think we need much fancy UI work here.

Reply:
"Skip content check for XPECT for extensions" doesn't seem like optimal wording.

"Suppress Xpect editor override for the following file extensions"? Not ideal but maybe better.